adapters/beachfront/beachfront.go

```diff
@@ -111,300 +113,334 @@ type BeachfrontResponseSlot struct {
-	var beachfrontReq = NewBeachfrontVideoRequest()
-	var errs = make([]error, 0, len(req.Imp))
-	var imps = 0
+		bfBannerRequest.Slots = append(bfBannerRequest.Slots, slot)
+	}
 
-	if req.App != nil {
-		if req.App.Domain != "" {
-			beachfrontReq.Site.Domain = req.App.Domain
-			beachfrontReq.Site.Page = req.App.ID
+	if request.Device != nil {
+		bfBannerRequest.IP = request.Device.IP
+		bfBannerRequest.DeviceModel = request.Device.Model
+		bfBannerRequest.DeviceOs = request.Device.OS
+		if request.Device.DNT != nil {
+			bfBannerRequest.Dnt = *request.Device.DNT
 		}
-	} else {
-		if req.Site.Page != "" {
-			if req.Site.Domain == "" {
-				if strings.Contains(req.Site.Page, "//") {
-					// Remove protocol if exists
-					beachfrontReq.Site.Domain = strings.Split(req.Site.Page, "//")[1]
-				}
-				if strings.Contains(beachfrontReq.Site.Domain, "/") {
-					// Drop everything after the first "/"
-					beachfrontReq.Site.Domain = strings.Split(beachfrontReq.Site.Domain, "/")[0]
-				}
-			} else {
-				beachfrontReq.Site.Domain = req.Site.Domain
-			}
-			beachfrontReq.Site.Page = req.Site.Page
+		if request.Device.UA != "" {
+			bfBannerRequest.UA = request.Device.UA
 		}
 	}
 
-	/*
-		The req could contain banner,audio,native and video imps when It arrives here. I am only
-		interested in video
+	site := getSite(request)
+	bfBannerRequest.IsMobile = site.Mobile
+	bfBannerRequest.Page = site.Page
+	bfBannerRequest.Domain = site.Domain
+
+	if request.User != nil {
+		if bfBannerRequest.User.ID == "" {
+			bfBannerRequest.User.ID = request.User.ID
+		}
 
-		The beachfront video endpoint is only capable of returning a single nurl and price, wrapped in
-		an openrtb format, so even though I'm building a request here that will include multiple video
-		impressions, only a single URL will be returned. Hopefully the beachfront endpoint can be modified
-		in the future to return multiple video ads
+		if bfBannerRequest.User.BuyerUID == "" {
+			bfBannerRequest.User.BuyerUID = request.User.BuyerUID
+		}
+	}
 
-	*/
-	for _, imp := range req.Imp {
-		if imp.Video != nil {
-			beachfrontReq.ID = req.ID
+	bfBannerRequest.RequestID = request.ID
 
-			beachfrontReq.Imp = append(beachfrontReq.Imp, BeachfrontVideoImp{})
-			videoImpsIndex = len(beachfrontReq.Imp) - 1
+	if request.Imp[0].Secure != nil {
+		bfBannerRequest.Secure = *request.Imp[0].Secure
+	}
 
-			beachfrontReq.Imp[videoImpsIndex].Video.H = imp.Video.H
-			beachfrontReq.Imp[videoImpsIndex].Video.W = imp.Video.W
+	return bfBannerRequest, errs
+}
 
-			var bidderExt adapters.ExtImpBidder
-			if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-				errs = append(errs, err)
-				continue
-			}
+/*
+getVideoRequests, plural. One request to the endpoint can have one appId, and can return one nurl,
+so each video imp is a call to the endpoint.
+*/
+func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []error) {
+	var beachfrontReqs = make([]beachfrontVideoRequest, 0)
```

Comment:
Please change the capacity for this slice to be `len(request.Imp)` as you expect to create that many video requests